src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -837,6 +837,42 @@ void C2_MacroAssembler::fast_unlock_lightweight(Register obj, Register reg_rax,
   // C2 uses the value of ZF to determine the continuation.
 }
 
+#ifdef ASSERT
+void C2_MacroAssembler::checked_cast_int(const TypeInt* type, Register dst) {
+  Label fail;
+  Label succeed;
+  cmpl(dst, type->_lo);
+  jccb(Assembler::less, fail);
+  cmpl(dst, type->_hi);
+  jccb(Assembler::lessEqual, succeed);
+  bind(fail);
+  stop("Invalid CastII");
```

Comment:
Is there any way to print the `[lo,hi]` and `dst`? I suppose one can always get that via the debugger, would just be convenient.

Reply:
Done, calling into VM somehow results in no stacktrace so I just call `hlt` instead.